dynamics/src/ParametricMesh.cpp

```diff
@@ -189,6 +189,80 @@ void ParametricMesh::readmesh(std::string fname)
     }
 }
 
+void ParametricMesh::coordinatesFromModelArray(const ModelArray& coords)
+{
+    // Fill in the array sizes from the ModelArray dimensions
+    nx = ModelArray::size(ModelArray::Dimension::X);
+    ny = ModelArray::size(ModelArray::Dimension::Y);
+    nelements = nx * ny;
+    nnodes = (nx + 1) * (ny + 1);
+    vertices.resize(nnodes, 2);
+    for (size_t idx = 0; idx < nnodes; ++idx) {
+        vertices(idx, 0) = coords.components(idx)[0];
+        vertices(idx, 1) = coords.components(idx)[1];
+    }
+}
+
+void ParametricMesh::landmaskFromModelArray(const ModelArray& mask)
+{
+    landmask.resize(nelements);
+    for (size_t idx = 0; idx < mask.trueSize(); ++idx)
+    {
+        landmask[idx] = (mask[idx] == 1.);
+    }
+}
+
+void ParametricMesh::dirichletFromMask()
+{
+    // BOTTOM, RIGHT, TOP, LEFT
```

Comment:
Sorry, I am confused here: should it be bottom left, bottom right, lop left, top right?

Reply:
This part deals with edges, not vertices. Here I am noting Thomas' order, as much for my benefit as anyone else's.

I have expanded the comment.

Reply:
Aaah, thanks